logllm/log_llm.py

```diff
@@ -1,44 +1,79 @@
+)
 
 def init_wandb(project_name):
     wandb.init(project=project_name, settings=wandb.Settings(_disable_stats=True))
 
-def extract_experimental_conditions(code):
+def extract_experimental_conditions_gemini(code):
+    user_input = f"""
+         You are an advanced machine learning experiment designer.
+         Extract all experimental conditions and results for logging via wandb API. 
+         Add your original parameters in your JSON response if you want to log other parameters.
+         Extract all information you can find in the given script as int, bool, or float values.
+         If you cannot describe conditions with int, bool, or float values, use a list of natural language.
+         Give advice to improve the accuracy.
+         If you use natural language, the answers should be short.
+         Do not include information already provided in param_name_1 for `condition_as_natural_language`.
+         
+         Here is a user's Jupyter Notebook script: {code}
+    """.replace("    ","")
+
+    chat_session = model.start_chat(
+        history=[
+            {"role": "user", "parts": ["Hello! help me analyse data in json format only"]},
+            {"role": "model", "parts": ["Sure I can do that, provide me with data"]},
+        ]
+    )
+
+    response = chat_session.send_message(user_input)
+    result = response.candidates[0].content.parts[0].text
+    
+    parsed_json = json.loads(result)
+    formatted_json = json.dumps(parsed_json, indent=4, ensure_ascii=False)
+    print("response: ", formatted_json)
+    return formatted_json
+
+def extract_experimental_conditions_openai(code):
     client = OpenAI()
 
     system_prompt = """
-        # You are advanced machine learning experiment designer.
-        # Extract all experimental conditions and results for logging via wandb api. 
-        # Add your original params in your JSON responce if you want to log other params.
-        # Extract all informaiton you can find the given script as int, bool or float value.
-        # If you can not describe conditions with int, bool or float value, use list of natural language.
-        # Give advice to improve the acc.
-        # If you use natural language, answer should be very short.
-        # Do not include information already provided in param_name_1 for `condition_as_natural_langauge`.
-        # Output JSON schema example:
-        This is just a example, make it change as you want. Use nested dictionally if nessasary.
-        {{
-            "method":"str",
-            "dataset":"str",
-            "task":"str",
-            "accuracy":"",
-            "other_param_here":{
-                "other_param_here":"",
-                "other_param_here":"",
-                },
-            "other_param_here":"",
-            ...
-            "condition_as_natural_langauge":["Small dataset."],
-            "advice_to_improve_acc":["Use bigger dataset.","Use more simple model."]
-        }}
```

Comment:
JSON is not necessary?

Reply:
Yes please check on the suggestions I made on the issues post.